lib/services/rocket.py

```diff
@@ -0,0 +1,295 @@
+                trigger_expression := parachute.trigger
+            ):
+                parachute.trigger = eval(  # pylint: disable=eval-used
+                    trigger_expression, {"__builtins__": None}, {}
+                )
+                rocketpy_parachute = cls.get_rocketpy_parachute(parachute)
+                rocketpy_rocket.parachutes.append(rocketpy_parachute)
+            else:
+                logger.warning(
+                    "Parachute trigger not valid. Skipping parachute."
+                )
+                continue
+
+        return cls(rocket=rocketpy_rocket)
+
+    @property
+    def rocket(self) -> RocketPyRocket:
+        return self._rocket
+
+    @rocket.setter
+    def rocket(self, rocket: RocketPyRocket):
+        self._rocket = rocket
+
+    def get_rocket_summary(self) -> RocketSummary:
+        """
+        Get the summary of the rocket.
+
+        Returns:
+            RocketSummary
+        """
+        attributes = get_instance_attributes(self.rocket)
+        rocket_summary = RocketSummary(**attributes)
+        return rocket_summary
+
+    def get_rocket_binary(self) -> bytes:
+        """
+        Get the binary representation of the rocket.
+
+        Returns:
+            bytes
+        """
+        return dill.dumps(self.rocket)
+
+    @staticmethod
+    def get_rocketpy_nose(nose: NoseCone) -> RocketPyNoseCone:
+        """
+        Get a rocketpy nose cone object.
+
+        Returns:
+            RocketPyNoseCone
+        """
+
+        rocketpy_nose = RocketPyNoseCone(
+            name=nose.name,
+            length=nose.length,
+            kind=nose.kind,
+            base_radius=nose.base_radius,
+            rocket_radius=nose.rocket_radius,
+        )
+        rocketpy_nose.position = nose.position
+        return rocketpy_nose
+
+    @staticmethod
+    def get_rocketpy_finset(fins: Fins, kind: str) -> RocketPyFins:
+        """
+        Get a rocketpy finset object.
+
+        Returns one of:
+            RocketPyTrapezoidalFins
+            RocketPyEllipticalFins
+        """
+        match kind:
+            case "TRAPEZOIDAL":
+                rocketpy_finset = RocketPyTrapezoidalFins(
+                    n=fins.n,
+                    name=fins.name,
+                    root_chord=fins.root_chord,
+                    tip_chord=fins.tip_chord,
+                    span=fins.span,
+                    cant_angle=fins.cant_angle,
+                    rocket_radius=fins.radius,
+                    airfoil=fins.airfoil,
+                )
+            case "ELLIPTICAL":
+                rocketpy_finset = RocketPyEllipticalFins(
+                    n=fins.n,
+                    name=fins.name,
+                    root_chord=fins.root_chord,
+                    span=fins.span,
+                    cant_angle=fins.cant_angle,
+                    rocket_radius=fins.radius,
+                    airfoil=fins.airfoil,
+                )
+            case _:
```

Comment:
why have you repeated the code for trapezoidal? suggestion to simply remove the first match condition

Reply:
Good point. I prefer to keep the good matches explicit tho, will raise an ValueError in case of default.

Patched in https://github.com/RocketPy-Team/Infinity-API/pull/25/commits/eace9ee9f611845855f10db7c7b302d2de589fa8